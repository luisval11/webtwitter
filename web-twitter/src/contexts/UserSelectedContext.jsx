import React from "react";

export const UserSelectedContext = React.createContext(null);
