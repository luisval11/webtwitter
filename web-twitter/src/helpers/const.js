export const DUMMY_POSTS = [
	{
		id: 1,
		authorId: 1,
		content: "Lorem ipsum content large",
		createdDate: new Date(2022, 2, 30, 14, 30),
	},
	{
		id: 2,
		authorId: 2,
		content: "Lorem ipsum content large 1",
		createdDate: new Date(2022, 2, 30, 10, 30),
	},
	{
		id: 3,
		authorId: 3,
		content: "Lorem ipsum content large 2",
		createdDate: new Date(2022, 2, 29, 10, 30),
	},
	{
		id: 4,
		authorId: 0,
		content: "Lorem ipsum content large 3",
		createdDate: new Date(2022, 2, 28, 10, 30),
	},
	{
		id: 5,
		authorId: 4,
		content: "Lorem ipsum content large 5",
		createdDate: new Date(2022, 2, 27, 10, 30),
	},
	{
		id: 6,
		authorId: 6,
		content: "Lorem ipsum content large 6",
		createdDate: new Date(2022, 2, 26, 9, 30),
	},
	{
		id: 7,
		authorId: 5,
		content: "Lorem ipsum content large 7",
		createdDate: new Date(2022, 2, 25, 12, 30),
	},
];

export const DUMMY_USERS = [
	{
		id: 1,
		name: "Alice",
		isFollowed: true,
	},
	{
		id: 2,
		name: "Bob",
		isFollowed: true,
	},
	{
		id: 3,
		name: "Mike",
		isFollowed: true,
	},
	{
		id: 4,
		name: "Beatriz",
		isFollowed: false,
	},
	{
		id: 5,
		name: "Ben",
		isFollowed: false,
	},
	{
		id: 6,
		name: "Jenny",
		isFollowed: false,
	},
];
