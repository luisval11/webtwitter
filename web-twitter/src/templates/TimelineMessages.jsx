import PropTypes from "prop-types";
import React, { useContext, useEffect, useState } from "react";
import { UserSelectedContext } from "../contexts/UserSelectedContext";
import { DUMMY_POSTS } from "../helpers/const";
import {
	getFilteredPosts,
	getFilteredPostsSelectedUser
} from "../helpers/utils/getFilteredPosts";
import { AddPostForm } from "../molecules/post/forms/AddPostForm";
import { List } from "../organisms/List";
import "./style.css";
export const TimelineMessages = ({ users }) => {
	const [posts, setPosts] = useState(DUMMY_POSTS);

	const [postsFiltered, setPostsFiltered] = useState([]);
	const { selectedUser } = useContext(UserSelectedContext);

	useEffect(() => {
		if (selectedUser === null) {
			const authorIds = users.map((user) => user.id);
			setPostsFiltered(getFilteredPosts(posts, authorIds, users));
		} else {
			setPostsFiltered(
				getFilteredPostsSelectedUser(posts, users, selectedUser)
			);
		}
	}, [selectedUser, posts, users]);

	return (
		<div className="timeline">
			<List title="Timeline" items={postsFiltered} type="post" />
			<AddPostForm setPosts={setPosts} />
		</div>
	);
};

TimelineMessages.propTypes = {
	users: PropTypes.array.isRequired,
};

export default TimelineMessages;
