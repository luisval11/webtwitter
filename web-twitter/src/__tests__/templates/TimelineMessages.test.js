import { render } from "@testing-library/react";
import { UserSelectedContext } from "../../contexts/UserSelectedContext";
import { DUMMY_USERS } from "../../helpers/const";
import TimelineMessages from "../../templates/TimelineMessages";

describe("Test on TimelineMessages", () => {
	const dummyUsers = DUMMY_USERS;

	test("should match initial snapshot for TimelineMessages", () => {
		const { container } = render(
			<UserSelectedContext.Provider value={{ selectedUser: null }}>
				<TimelineMessages users={dummyUsers} />
			</UserSelectedContext.Provider>
		);
		expect(container).toMatchSnapshot();
	});

	test("should have 6 initial div elements with className 'list-group-item'", () => {
		const { container } = render(
			<UserSelectedContext.Provider value={{ selectedUser: null }}>
				<TimelineMessages users={dummyUsers} />
			</UserSelectedContext.Provider>
		);
		const messagesLenght =
			container.getElementsByClassName("list-group-item").length;
		expect(messagesLenght).toBe(6);
	});
});
