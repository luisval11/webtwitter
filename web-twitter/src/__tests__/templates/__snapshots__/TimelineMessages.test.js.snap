// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on TimelineMessages should match initial snapshot for TimelineMessages 1`] = `
<div>
  <div
    class="timeline"
  >
    <div
      class="list"
    >
      <h6>
        Timeline
      </h6>
      <div
        aria-labelledby="post"
        class="list-group"
        role="list"
      >
        <div
          class="list-item list-group-item"
          role="listitem"
        >
          <div
            class="post card"
          >
            <div
              class="card-body"
            >
              <div
                class="post--title"
              >
                <div
                  class="m-0 card-title h5"
                >
                  Alice
                </div>
                <span>
                  30 mar
                </span>
              </div>
              <p
                class="mb-0 card-text"
              >
                Lorem ipsum content large
              </p>
            </div>
          </div>
        </div>
        <div
          class="list-item list-group-item"
          role="listitem"
        >
          <div
            class="post card"
          >
            <div
              class="card-body"
            >
              <div
                class="post--title"
              >
                <div
                  class="m-0 card-title h5"
                >
                  Bob
                </div>
                <span>
                  30 mar
                </span>
              </div>
              <p
                class="mb-0 card-text"
              >
                Lorem ipsum content large 1
              </p>
            </div>
          </div>
        </div>
        <div
          class="list-item list-group-item"
          role="listitem"
        >
          <div
            class="post card"
          >
            <div
              class="card-body"
            >
              <div
                class="post--title"
              >
                <div
                  class="m-0 card-title h5"
                >
                  Mike
                </div>
                <span>
                  29 mar
                </span>
              </div>
              <p
                class="mb-0 card-text"
              >
                Lorem ipsum content large 2
              </p>
            </div>
          </div>
        </div>
        <div
          class="list-item list-group-item"
          role="listitem"
        >
          <div
            class="post card"
          >
            <div
              class="card-body"
            >
              <div
                class="post--title"
              >
                <div
                  class="m-0 card-title h5"
                >
                  Beatriz
                </div>
                <span>
                  27 mar
                </span>
              </div>
              <p
                class="mb-0 card-text"
              >
                Lorem ipsum content large 5
              </p>
            </div>
          </div>
        </div>
        <div
          class="list-item list-group-item"
          role="listitem"
        >
          <div
            class="post card"
          >
            <div
              class="card-body"
            >
              <div
                class="post--title"
              >
                <div
                  class="m-0 card-title h5"
                >
                  Jenny
                </div>
                <span>
                  26 mar
                </span>
              </div>
              <p
                class="mb-0 card-text"
              >
                Lorem ipsum content large 6
              </p>
            </div>
          </div>
        </div>
        <div
          class="list-item list-group-item"
          role="listitem"
        >
          <div
            class="post card"
          >
            <div
              class="card-body"
            >
              <div
                class="post--title"
              >
                <div
                  class="m-0 card-title h5"
                >
                  Ben
                </div>
                <span>
                  25 mar
                </span>
              </div>
              <p
                class="mb-0 card-text"
              >
                Lorem ipsum content large 7
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <form
      class=""
    >
      <div
        class="input__textarea"
      >
        <label
          class="form-label"
          for="content"
        >
          Post a new message
        </label>
        <textarea
          class="form-control"
          id="content"
          role="textarea"
        />
      </div>
      <div
        class="form__button--submit"
      >
        <button
          class="btn btn-primary"
          type="button"
        >
          Post
        </button>
      </div>
    </form>
  </div>
</div>
`;
